test/UnitTests/Microsoft.OData.Core.Tests/ScenarioTests/UriBuilder/QueryOptionBuilderTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="QueryOptionBuilderTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using Xunit;
+
+namespace Microsoft.OData.Tests.ScenarioTests.UriBuilder
+{
+    public class QueryOptionBuilderTests : UriBuilderTestBase
+    {
+        [Fact]
+        public void CustomQueryOptionWorks()

```

Comment:
shall we have failing test case?
for example, only contain partial, `customerQueryOption=`